arch/risc-v/include/rv32m1/rv32m1ri5cy_irq.h

```diff
@@ -0,0 +1,126 @@
+/****************************************************************************
+ * arch/risc-v/include/rv32m1/rv32m1ri5cy_irq.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_RISCV_INCLUDE_RV32M1_RV32M1RI5CY_IRQ_H
+#define __ARCH_RISCV_INCLUDE_RV32M1_RV32M1RI5CY_IRQ_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Map RISC-V exception code to NuttX IRQ */
+
+/* IRQ 0-15 : (exception:interrupt=0) */
+
+#define RV32M1_IRQ_IAMISALIGNED  (0) /* Instruction Address Misaligned */
+#define RV32M1_IRQ_IAFAULT       (1) /* Instruction Address Fault */
+#define RV32M1_IRQ_IINSTRUCTION  (2) /* Illegal Instruction */
+#define RV32M1_IRQ_BPOINT        (3) /* Break Point */
+#define RV32M1_IRQ_LAMISALIGNED  (4) /* Load Address Misaligned */
+#define RV32M1_IRQ_LAFAULT       (5) /* Load Access Fault */
+#define RV32M1_IRQ_SAMISALIGNED  (6) /* Store/AMO Address Misaligned */
+#define RV32M1_IRQ_SAFAULT       (7) /* Store/AMO Access Fault */
+#define RV32M1_IRQ_ECALLU        (8) /* Environment Call from U-mode */
+                                    /* 9-10: Reserved */
+
+#define RV32M1_IRQ_ECALLM       (11) /* Environment Call from M-mode */
+                                    /* 12-15: Reserved */
```

Comment:
Why don't you use the definition from arch/risc-v/include/rv32im/mcause.h ? It seams to be standard for RISC-V